edx_notifications/stores/sql/tests/test_store_provider.py

```diff
@@ -865,3 +902,183 @@ def test_get_nonexisting_timer(self):
 
         with self.assertRaises(ItemNotFoundError):
             self.provider.get_notification_timer('foo')
+
+    def test_save_notification_preference(self):
+        """
+        test save notification preference in the store provide.
+        """
+        notification_preference = self._save_notification_preference(
+            name='test_notification_preference',
+            display_name="Test Preference",
+            display_description="This is the test preference"
+        )
+
+        notification_preference_read = self.provider.get_notification_preference(
```

Comment:
when we typically test these database accessor methods, we wrap these calls in a self.assertNumQueries() - see above examples - to make sure that the database is only hit N times. This is to catch situations where there might be a performance degradation due to a change. This might take a few times to figure out what the right number of N is (e.g. sometimes create-or-update operations are 2 database calls) for every call site, but you can do:

```
with self.assertNumQueries(1):
    ....some call to store().....
```

And then take the number that the assertion raises in the console output and update the assertion count.